### ISSUE TITLE
cuff to short or wrist part of sleeve to large on simon 

### ISSUE THREAD

**Issue by grimlokason** [freesewing/freesewing#253]
Hello,

It seems that the cuff is too short when setted to barrel cuff to the wrist part of the sleeve + the button placket.
For exemple in : https://freesewing.org/recreate/falyi/for/nefri it's 18,75 when setted to barrel ( button to hole : 16,5 ) , and 20,63cm if it's french cuff (button to hole : 17,25 )

NB : the wrist measurment was 15cm and the ease was 10% so i assume it's there is something wrong on the sleeve calcul ?

NB bis, i'm not sure, but it's probably here : 
```java
// Sleeve width depends on cuff style
  let width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap)
  if (
    options.cuffStyle === 'straightFrenchcuff' ||
    options.cuffStyle === 'roundedFrenchcuff' ||
    options.cuffStyle === 'angledFrenchCuff'
  )
    width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap * 1.5)
```
Wich should become : 
```java
// Sleeve width depends on cuff style
  let width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap)
  if (
    options.cuffStyle === 'straightFrenchcuff' ||
    options.cuffStyle === 'roundedFrenchcuff' ||
    options.cuffStyle === 'angledFrenchCuff'
  ) {
    width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap * 1.5)
}
```

Best regards

**Comment by joostdecock** [comment]
From what you describe, I don't think there's a problem.
Your wrist circumference is 15cm, and you asked for 10% ease. 10% of 15cm is 1.5cm so the cuff should be 15+1.5 = 16.5cm. You say yourself that the barrel cuff button to buttonhole = 16.5cm which means it's exactly what we'd expect it to be.

Perhaps the issue is that the cuff is shorter than the bottom of the sleeve? That's because there's folds at the bottom of the sleeve. This is always the case on a shirt, and the exact amount is determined by the **cuff drape** option.

Or perhaps you feel that 10% ease is not enough. But the default cuff ease is 20%, so you have lowered it to half of that.

Please clarify if I'm somehow misunderstanding your issue.

**Comment by grimlokason** [comment]
Hello, 
Sorry, i've been doing more test and analyse while writting the issue and i was thinking that the default was in the cuff part at start, that's why it's not well explained >.<

From what i understand, if we set the cuff to barrel cuff, the bottom of the sleeve should be : 
width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap) ( so a little more than 18,75 since we have the 2 folds )
Instead, i think we got the french cuff settings  : 
width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap * 1.5)
since 

If we try to switch from french cuff to barrel cuff, the bottom of the sleeve and the cuff should change, but, there is just the cuff that change,  ( probably because of the missing { } after the if in https://github.com/freesewing/freesewing/blob/develop/packages/simon/src/sleeve.js wich made the  width  always set by the line  32 ) 

Best regards.

**Comment by joostdecock** [comment]
Congratulations, you found a bug :) but it's not what you think it is. Allow me to explain:

> If we try to switch from french cuff to barrel cuff, the bottom of the sleeve and the cuff should change, but, there is just the cuff that change, ( probably because of the missing { } after the if in https://github.com/freesewing/freesewing/blob/develop/packages/simon/src/sleeve.js wich made the width always set by the line 32 )

You are correct that the sleeve width should change. But it's not because of the missing `{` and `}`.
In JavaScript, an `if` doesn't require curly braces if there's only 1 line inside the block guarded by the if statement.

For example, all these code snippets have the same result:

```js
if (this) {
 do('that')
}

if (this) do('that')

if (this)
do ('that')
```

So this:

```js
if (
    options.cuffStyle === 'straightFrenchcuff' ||
    options.cuffStyle === 'roundedFrenchcuff' ||
    options.cuffStyle === 'angledFrenchCuff'
  )
    width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap * 1.5)
``` 

is the same as the fix you suggest:

```js
if (
    options.cuffStyle === 'straightFrenchcuff' ||
    options.cuffStyle === 'roundedFrenchcuff' ||
    options.cuffStyle === 'angledFrenchCuff'
  ) {
    width = measurements.wristCircumference * (1 + options.cuffEase + options.cuffOverlap * 1.5)
}
```

But, there is still a bug in this code. Two to be precise. The problem is that `straightFrenchcuff` and `straightFrenchcuff` do not have a capital C for cuff. `angledFrenchCuff` does, and you'll notice that if you pick an angled French cuff, the sleeve width does update as expected.

To fix this, we need to:

 - change straightFrenchcuff into straightFrench**C**uff [on line 28](https://github.com/freesewing/freesewing/blob/develop/packages/simon/src/sleeve.js#L28)
 - change roundedFrenchcuff to roundedFrench**C**uff [on line 29](https://github.com/freesewing/freesewing/blob/develop/packages/simon/src/sleeve.js#L29)


Since you went through the trouble of going through the code to understand what's going on, perhaps you would like to submit a pull request for this?

For that reason, I'll hold off fixing this myself for now.

Thank you for reporting this, much appreciated!

**Comment by joostdecock** [comment]
Fixed in v2.1.9
